reactornettybenchmark/src/test/java/com/ss/benchmark/httpclient/reactornetty/ReactorNettyHC.java

```diff
@@ -0,0 +1,120 @@
+package com.ss.benchmark.httpclient.reactornetty;
+
+import com.ss.benchmark.httpclient.HC;
+import io.netty.channel.ChannelOption;
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.timeout.ReadTimeoutHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.netty.ByteBufFlux;
+import reactor.netty.http.client.HttpClient;
+import reactor.netty.resources.ConnectionProvider;
+
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+public class ReactorNettyHC implements HC {
+    private static final Logger logger = LoggerFactory.getLogger(ReactorNettyHC.class);
+
+    HttpClient client;
+
+    @Override
+    public void setup(String baseURL) {
+        client = HttpClient
+                .create(ConnectionProvider.fixed("benchmark", MAX_CONNECTION_POOL_SIZE, CONNECTION_REQUEST_TIMEOUT))
+                .baseUrl(baseURL)
+                .tcpConfiguration(tcpClient ->
+                        tcpClient.option(ChannelOption.CONNECT_TIMEOUT_MILLIS, CONNECT_TIMEOUT)
+                                .doOnConnected(con -> con.addHandlerLast(new ReadTimeoutHandler(READ_TIMEOUT, TimeUnit.MILLISECONDS))));
+    }
+
+    @Override
+    public String blockingGET(String uri) {
+        HttpClient.RequestSender requestSender = client
+                .request(HttpMethod.GET)
+                .uri(uri);
+
+        return requestSender
+                .responseSingle((res, body) -> {
+                    if (res.status().code() != 200) {
+                        Mono.error(new IllegalStateException("Unexpected response code : " + res.status().code()));
+                    }
+                    return body;
+                })
+                .map(byteBuf -> byteBuf.toString(StandardCharsets.UTF_8))
+                .doOnError(t -> System.err.println("Failed requesting server: " + t.getMessage()))
+                .block();
```

Comment:
There is code duplication in this class that could be removed.  Not a big deal though.  If we merge and I feel like it, I'll refactor it:)

Reply:
let's leave for now